### DISCUSSION TITLE
Is there a project example bigger than a single file?

### DISCUSSION THREAD

**Discussion by kkweon** [google/wire#306] [General]
Is there an example that goes beyond 1 single file?

What's the _recommended_ structure?

I know it all depends on the use cases and wire doesn't force users obviously.
But, I'd still like to see an example that shows how it would look in a project beyond just a single file.

What's the convention for file names and how to split them ?

I often required to create `wire.go` and multiple `wire.NewSets(...)` per package, mainly because I can use in tests. For example, I've been creating `package_set.go` per each package that contains multiple `wire.NewSets(...)` and `wire.go` per package for testing.

**Comment by cglotr** [comment]
I use Wire in one of my side projects [here](https://github.com/arikama/koran-backend/blob/master/wire.go). It's a fairly small codebase but I don't think it will change much if the codebase gets bigger

**Comment by batazor** [comment]
I use wire for my monorepo - https://github.com/shortlink-org/shortlink/blob/main/internal/services/link/di/wire.go
